flake8_idom_hooks/utils.py

```diff
@@ -28,12 +28,18 @@ def is_hook_def(node: ast.FunctionDef) -> bool:
 
 
 def is_component_def(node: ast.FunctionDef) -> bool:
-    return is_component_function_name(node.name)
-
-
-def is_component_function_name(name: str) -> bool:
-    return name[0].upper() == name[0] and "_" not in name
+    return any(
+        decorator.value.id == "idom" and decorator.attr == "component"
+        for decorator in node.decorator_list
+    )
 
 
 def is_hook_function_name(name: str) -> bool:
-    return name.lstrip("_").startswith("use_")
```

Comment:
This rule cannot be changed because users may define their own hooks. There is no way for the linter to know what a "hook" is without some sort of naming convention.

Reply:
How would you feel about `var, var = use_*` equating to a hook? More specific and less likely to match to other user generated stuff.

Reply:
Fortunately or unfortunately hooks aren't limited to that one form. React's own linter uses this same convention - if there were a better way to statically identify hooks I suspect they would have found it.